src/device-api/v1.js

```diff
@@ -0,0 +1,126 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as constants from '../lib/constants';
+import { checkInt, checkTruthy } from '../lib/validation';
+import { doRestart, doPurge, serviceAction } from './common';
+
+export const createV1Api = function(router, applications) {
+	const { eventTracker } = applications;
+
+	router.post('/v1/restart', function(req, res, next) {
+		const appId = checkInt(req.body.appId);
+		const force = checkTruthy(req.body.force) ?? false;
+		eventTracker.track('Restart container (v1)', { appId });
+		if (appId == null) {
+			return res.status(400).send('Missing app id');
+		}
+		return doRestart(applications, appId, force)
+			.then(() => res.status(200).send('OK'))
+			.catch(next);
+	});
+
+	const v1StopOrStart = function(req, res, next, action) {
+		const appId = checkInt(req.params.appId);
+		const force = checkTruthy(req.body.force) ?? false;
```

Comment:
This `?? false` wasn't technically required by typescript but it seemed the right thing to do based on the two cases it did catch

Reply:
Yeah, the `checkTruthy` function is a mess, but I'd like the entire codebase to be in typescript before changing fundamentals like that.